graphql/tests/query.rs

```diff
@@ -2123,3 +2123,46 @@ fn deterministic_error() {
         assert_eq!(expected, serde_json::to_value(&result).unwrap());
     })
 }
+
+#[test]
+fn can_query_with_or_filter() {
+    const QUERY: &str = "
+    query {
+        musicians(where: { or: { name: \"John\", id: \"m2\" } }) {
+            name
+            id
+        }
+    }
+    ";
+
+    run_query(QUERY, |result, _| {
+        let exp = object! {
+            musicians: vec![
+                object! { name: "John", id: "m1" },
+                object! { name: "Lisa", id: "m2" },
+            ],
+        };
+        let data = extract_data!(result).unwrap();
+        assert_eq!(data, exp);
+    })
+}
+
+#[test]
+fn can_query_with_and_filter() {
+    const QUERY: &str = "
+    query {
+        musicians(where: { and: { name: \"John\", id: \"m2\" } }) {
+            name
+            id
+        }
+    }
+    ";
+
+    run_query(QUERY, |result, _| {
+        let exp = object! {
+            musicians: r::Value::List(vec![]),
+        };
+        let data = extract_data!(result).unwrap();
+        assert_eq!(data, exp);
+    })
+}
```

Comment:
It would be good to also have some tests that mix `and` and `or` filters just to make sure that doesn't get broken inadvertently in the future. Maybe something like `where: { or: { and: { name: "John", id: "m1" }, and: { name: "Lisa", id: "m1" }, { mainBand: "b2" } } }` which, AFAICT, should return `John` and `Tom`

Reply:
added 2 more nested tests